ui/src/variables/components/VariableArgumentsEditor.tsx

```diff
@@ -21,20 +22,21 @@ interface Props {
   onChange: (update: {args: VariableArguments; isValid: boolean}) => void
   onSelectMapDefault: (selectedKey: string) => void
   selected: string[]
+  variableType: VariableArgumentType
 }
 
 class VariableArgumentsEditor extends PureComponent<Props> {
   render() {
-    const {args, onSelectMapDefault, selected} = this.props
+    const {args, onSelectMapDefault, selected, variableType} = this.props
 
-    switch (args.type) {
+    switch (variableType) {
       case 'query':
         return (
           <Form.Element label="Script">
             <Grid.Column>
               <div className="overlay-flux-editor">
                 <FluxEditor
-                  script={args.values.query}
+                  script={(args as QueryArguments).values.query}
```

Comment:
Generally a smell when I have to cast a type using `as`

Reply:
felt the same way, seemed to come along with the use of the generics and the `vartype = type1 | type2 | type3` definitions. typescript seems to be only able to resolve in one direction. any tips on mitigating this kind of polymorphism?